core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -1815,15 +1817,59 @@ public void onSetPlaybackSpeed(float speed) {
             setSpeed(speed);
         }
 
+        private boolean playStateBeforeClick = false;
+        private long lastClickTime = 0;
+        private Handler clickHandler = new Handler(Looper.getMainLooper());
+        private int clickCount = 0;
+
+        private Runnable clickRunnable = new Runnable() {
+            @Override
+            public void run() {
+                if (clickCount == 1) {
+                    if (playStateBeforeClick) {
+                        onPause();
+                    } else {
+                        onPlay();
+                    }
+                } else if (clickCount == 2) {
+                    onFastForward();
+                } else if (clickCount == 3) {
+                    onRewind();
+                    onPlay(); // otherwise playback is paused?
+                }
+                // Reset
+                clickCount = 0;
+            }
+        };
+
         @Override
         public boolean onMediaButtonEvent(final Intent mediaButton) {
             Log.d(TAG, "onMediaButtonEvent(" + mediaButton + ")");
             if (mediaButton != null) {
-                KeyEvent keyEvent = mediaButton.getParcelableExtra(Intent.EXTRA_KEY_EVENT);
-                if (keyEvent != null &&
-                        keyEvent.getAction() == KeyEvent.ACTION_DOWN &&
-                        keyEvent.getRepeatCount() == 0) {
-                    return handleKeycode(keyEvent.getKeyCode(), false);
+                final KeyEvent keyEvent = mediaButton.getParcelableExtra(Intent.EXTRA_KEY_EVENT);
+                if (keyEvent != null && keyEvent.getRepeatCount() == 0) {
+                    // Check if the key event is a play/pause button to avoid other button presses
+                    if (keyEvent.getKeyCode() == KeyEvent.KEYCODE_MEDIA_PLAY_PAUSE
+                            || keyEvent.getKeyCode() == KeyEvent.KEYCODE_HEADSETHOOK) {
+                        if (keyEvent.getAction() == KeyEvent.ACTION_DOWN) {
+                            long clickTime = System.currentTimeMillis();
+                            long elapsedTime = clickTime - lastClickTime;
+                            if (elapsedTime <= 500) { // Assuming 500 milliseconds threshold for multiple clicks
+                                clickCount++;
+                            } else {
+                                clickCount = 1; // if more than 500ms has passed, reset to 1
+                            }
```

Comment:
Doesn't the handler already do this? Why is this additional check needed?

Reply:
The Handler doesn't track the number of times the button has been pressed; it only schedules a Runnable to execute after a delay. So we need to maintain a separate count of button presses so that clickRunnable can use it to decide what action to take.

Reply:
The handler resets the count to 0 after the time threshold, so the threshold does not need to be checked again and reset again here.